src/packagers/JSPackager.js

```diff
@@ -93,6 +93,17 @@ class JSPackager extends Packager {
     return name;
   }
 
+  dedupeKey(asset) {
+    // cannot rely *only* on generated JS for deduplication because paths like
+    // `../` can cause 2 identical JS files to behave differently depending on
+    // where they are located on the filesystem
+    let deps = Array.from(asset.depAssets.keys(), dep =>
+      this.bundler.resolver.resolveFilename(dep.name, asset.name)
```

Comment:
if you use `depAssets.values()`, you can just map to `dep.name` which is already resolved